app/lib/activity_stream_reader.rb

```diff
@@ -47,20 +47,9 @@ def relevant?(item)
   def find_by_id(item)
     parsed_identifier_uri = parse_identifier_uri(item)
     metadata_source = parsed_identifier_uri[0]
-    source_id_type = parsed_identifier_uri[1]
-    source_id = parsed_identifier_uri[2]
-    if metadata_source == "ladybird" || metadata_source == "ils" # aka Voyager
-      oid = ParentObject.where(source_id_type.to_s => source_id.to_s)&.first&.oid
-      return false unless oid
-      oids_for_update.add([oid, metadata_source])
-    elsif metadata_source == "aspace"
-      source_id = ("/" + source_id_type + "/" + source_id).to_s
-      oid = ParentObject.where("aspace_uri" => source_id.to_s)&.first&.oid
-      return false unless oid
-      oids_for_update.add([oid, metadata_source])
-    else
-      return false
-    end
+    oid = DependentObject.where(dependent_uri: parse_dependent_uri(item))&.first&.parent_object_id&.to_s
```

Comment:
find_by_id makes me think:
1) I pass in an id (not an item)
2) I get back either an object or nil

this is mostly because it is such an "active recordy" name. That doesn't seem to be what this method does at this point, it returns true or false now.

also it seems like this still may need to be a loop according to @martinlovell comments? everything else looks good to me now.